src/Handler/StreamHandler.php

```diff
@@ -428,7 +430,24 @@ function ($code, $a, $b, $c, $transferred, $total) use ($value) {
         );
     }
 
-    private function add_debug(RequestInterface $request, &$options, $value, &$params)
+    private function add_force_ip_resolve(RequestInterface $request, &$options, $value, &$params, UriInterface &$uri)
+    {
+        if ('v4' === $value) {
+            $records = dns_get_record($uri->getHost(), DNS_A);
+            if (!isset($records[0]['ip'])) {
+                throw new ConnectException("Could not resolve IPv4 address for host '{$uri->getHost()}'", $request);
+            }
+            $uri = $uri->withHost($records[0]['ip']);
```

Comment:
I'm not sure what you mean. I am not modifying the HTTP message here, only $uri that is passed by reference.